src/ios/AppDelegate+notification.m

```diff
@@ -182,49 +267,316 @@ - (void)pushPluginOnApplicationDidBecomeActive:(NSNotification *)notification {
+                        }
+                        
+                    }
+
+                }
+                    
+            }
+            
+            // Remove all notifications that are waiting in the notification center:
+            // - setting a 0 value to remove the badge number of the app.
+            [[UIApplication sharedApplication] setApplicationIconBadgeNumber:0];
+            // - Remove all delivered notifications (items displayed in the notification center of the phone).
+            [[UNUserNotificationCenter currentNotificationCenter] removeAllDeliveredNotifications];
+            // - Remove all pending notifications which are not delivered yet but scheduled (like the local notification created by us to open all notifications at the same time).
+            [[UNUserNotificationCenter currentNotificationCenter] removeAllPendingNotificationRequests];
+            
+            // Create a single string with all notification IDs, concatenating them with a comma.
+            NSString *notificationsToOpen = [notificationIDlistToOpen componentsJoinedByString:@","];
+            
+            // Prepare the data to send the notification data to the CORES Mobie app.
+            pushHandler.isTapped = YES;
+            pushHandler.coldstart = [self.coldstart boolValue];
+            pushHandler.notificationIDsToOpen = notificationsToOpen;
+            pushHandler.notificationsGroupedContentList = @"YES";
+            [pushHandler notificationReceived];
+
+        }];
+        
+    }
+
+}
+
+// Method invoked to verify if the Cordova Plugin Push is ready to invoke the method used to send data to the CORES Mobile app.
+- (void) onPluginReady:(NSTimer *)timer {
+    
+    // Get from the timer the response data of the pressed notification (the method who invoked this function, check the "didReceiveNotificationResponse" method ).
+    NSDictionary *userInfo = [timer userInfo];
+    
+    // Initialize the PushPlugin class of the Cordova Push plugin and check if it was initialized correctly (checking its ID auto assigned by
+    // Cordova). If it is nil that means the Cordova plugins are not ready to be invoked (like for example: when the app was opened from scratch and the plugins are being initialized).
+    PushPlugin *pushHandler = [self getCommandInstance:@"PushNotification"];
+    if(pushHandler.callbackId != nil){
+        
+        // If the Cordova Push plugin is ready then we can finish the timer and proceed to send the data of the pressed notification to CORES Mobile JS logic.
+        if(checkPluginReadyTimer != nil){
+            [checkPluginReadyTimer invalidate];
+            checkPluginReadyTimer = nil;
+        }
+        
+        [self sendResponseDataOfPressedNotification : userInfo : pushHandler];
+        
+    }
+}
+
+// Method used to create a local/scheduled notification that will be displayed at the top of all app notifications to let the user open all of them in the Inbox page.
```

Comment:
Just FYI, this also looks like something that we would want to exclude from the pull request to the upstream repo.